runtime/descriptor.go

```diff
@@ -98,24 +105,54 @@ func propertySet(f *Frame, desc, inst, value *Object) *BaseException {
 
 // makeStructFieldDescriptor creates a descriptor with a getter that returns
 // the field given by fieldName from t's basis structure.
-func makeStructFieldDescriptor(t *Type, fieldName, propertyName string) *Object {
+func makeStructFieldDescriptor(t *Type, fieldName, propertyName string, fieldType fieldDescriptorType) *Object {
```

Comment:
AFAICT canSet is not available on the field, only the value. Since the value is only available in the `__set__` call, not during creation of the descriptor, the best we could do is a runtime check and raise an exception. That's better than panicking, but I think it's better to logFatal during Grumpy package initialization so that we can detect the error as early as possible.